tests/scripts/all.sh

```diff
@@ -4136,6 +4136,43 @@ component_check_test_helpers () {
     python3 -m unittest tests/scripts/translate_ciphers.py 2>&1
 }
 
+component_test_aes_builtin_only () {
```

Comment:
Question - do we also have tests that cover runtime detection (e.g., `MBEDTLS_AESNI_C` set AND `MBEDTLS_AES_DONT_USE_SOFTWARE_CRYPTO` unset)? We need to test run-time detection for all platforms, as it is different for each and would likely break independently.

Reply:
No, we don't have such tests yet. See https://github.com/Mbed-TLS/mbedtls/issues/7430#issuecomment-1505015489